core/client/fs/src/test/java/alluxio/client/file/FileInStreamTest.java

```diff
@@ -310,6 +310,38 @@ public void longSeekBackwardCachingPartiallyReadBlocks() throws IOException {
   }
 
   /**
+   * Tests reading and seeking with no local worker. Nothing should be cached.
+   */
+  @Test
+  public void testSeekWithNoLocalWorker() throws IOException {
+    // Overrides the get local worker call
+    PowerMockito.when(mContext.getLocalWorker()).thenReturn(null);
+    mTestStream =
+        new FileInStream(mStatus, InStreamOptions.defaults().setCachePartiallyReadBlock(true)
+            .setReadType(ReadType.CACHE_PROMOTE).setSeekBufferSizeBytes(7), mContext);
+    int readAmount = (int) (BLOCK_LENGTH / 2);
+    byte[] buffer = new byte[readAmount];
+    // read and seek several times
+    mTestStream.read(buffer);
+    Assert.assertEquals(readAmount, mInStreams.get(0).getBytesRead());
+    mTestStream.seek(BLOCK_LENGTH + BLOCK_LENGTH / 2);
+    mTestStream.seek(0);
+
+    if (mBlockSource == BlockInStreamSource.UFS) {
+      // reads the entire block to partial cache on the remote worker
+      Assert.assertEquals(BLOCK_LENGTH, mInStreams.get(0).getBytesRead());
+      Assert.assertEquals(BLOCK_LENGTH, mInStreams.get(1).getBytesRead());
+    } else {
+      // only reads the read amount
+      Assert.assertEquals(readAmount, mInStreams.get(0).getBytesRead());
+      Assert.assertEquals(0, mInStreams.get(1).getBytesRead());
+    }
+
+    // nothing is written to cache stream
+    Assert.assertEquals(0, mCacheStreams.get(0).getWrittenData().length);
+  }
```

Comment:
Can we add a test where we read from a remote worker w/ a local worker and verify the partial & passive cache behavior?
Can we add a test where we read from a remote worker from UFS w/o a local worker and verify the partial cache behavior?

Reply:
It already has these tests.

This test is parameterized on the source types including `local,remote,UFS` with a local worker.